delira/utils/config.py

```diff
@@ -1,5 +1,142 @@
+                except (KeyError, ValueError):
+                    current_level = current_level[k]
+            try:
+                return current_level[int(final_key)]
+            except (KeyError, ValueError):
+                return current_level[final_key]
+
+    def __contains__(self, key):
+        if not isinstance(key, str) or '.' not in key:
+            return super().__contains__(key)
+        else:
+            current_level = self
+            key_split = key.split(".")
+            final_key = key_split.pop(-1)
+            for k in key_split:
+                # traverse to needed dict
+                if isinstance(current_level, dict) or k not in current_level:
+                    return False
+                else:
+                    current_level = current_level[k]
+            return (final_key in current_level)
+
+    @staticmethod
+    def create_from_dict(value, val_copy=False):
+        assert isinstance(value, dict)
+        new_config = BaseConfig()
+        for key, item in value.items():
+            if isinstance(item, dict):
+                item = BaseConfig.create_from_dict(item, val_copy=val_copy)
+            if val_copy:
+                new_config[key] = copy.deepcopy(item)
+            else:
+                new_config[key] = item
+        return new_config
+
+    def update(self, update_dict, val_copy=False, overwrite=False):
+        for key, item in update_dict.items():
+            if key in self:
+                if isinstance(item, dict):
+                    self[key].update(update_dict[key], val_copy=val_copy,
+                                     overwrite=overwrite)
+                else:
+                    if overwrite:
+                        if val_copy:
+                            self[key] = copy.deepcopy(item)
+                        else:
+                            self[key] = item
+                    else:
+                        raise ValueError("{} already in config. Can "
+                                         "not overwrite value.".format(key))
+            else:
+                if isinstance(item, dict) and not isinstance(item, BaseConfig):
+                    self[key] = BaseConfig.create_from_dict(
+                        item, val_copy=val_copy)
+                else:
+                    if val_copy:
+                        self[key] = copy.deepcopy(item)
+                    else:
+                        self[key] = item
+
+    # TODO: support for json, yaml
+    # TODO: support for saving complex objects
+    def dump():
+        raise NotImplementedError
+
+    def dumps():
+        raise NotImplementedError
+
+    # TODO: support for json, yaml and argparse
+    # TODO: support for loading complex objects
+    def load():
+        raise NotImplementedError
+
+    def loads():
+        raise NotImplementedError
+
+    # TODO: check if copy and deepcopy works out of the box
+    def __copy__(self, *args, **kwargs):
```

Comment:
This probably won't work, since it says for me, that `dict` has no such attribute, maybe leave this one out or implement it from scratch